src/background/services/secrets/SecretsService.ts

```diff
@@ -477,4 +522,291 @@ export class SecretsService {
+    const walletSecrets = await this.getWalletAccountsSecretsById(id);
+    return walletSecrets?.secretType;
+  }
+
+  async addImportedWallet(importData: ImportData, isMainnet: boolean) {
+    const id = crypto.randomUUID();
+
+    // let the AccountService validate the account's uniqueness and save the secret using this callback
+    const commit = async () => {
+      // Need to re-map `data` to `secret` for private key imports
+      switch (importData.importType) {
+        case ImportType.PRIVATE_KEY:
+          await this.saveImportedWallet(id, {
+            secretType: SecretType.PrivateKey,
+            secret: importData.data,
+          });
+          break;
+        case ImportType.FIREBLOCKS:
+          await this.saveImportedWallet(id, {
+            secretType: SecretType.Fireblocks,
+            ...importData.data,
+          });
+          break;
+        case ImportType.WALLET_CONNECT:
+          await this.saveImportedWallet(id, {
+            secretType: SecretType.WalletConnect,
+            ...importData.data,
+          });
+          break;
+      }
+    };
+
+    if (
+      importData.importType === ImportType.FIREBLOCKS ||
+      importData.importType === ImportType.WALLET_CONNECT
+    ) {
+      return {
+        account: {
+          id,
+          ...importData.data.addresses,
+        },
+        commit,
+      };
+    }
+
+    if (importData.importType === ImportType.PRIVATE_KEY) {
+      return {
+        account: {
+          id,
+          ...this.#calculateAddressesForPrivateKey(importData.data, isMainnet),
+        },
+        commit,
+      };
+    }
+
+    throw new Error('Unknown import type');
+  }
+
+  #calculateAddressesForPrivateKey(privateKey: string, isMainnet: boolean) {
+    const addresses = {
+      addressBTC: '',
+      addressC: '',
+      addressAVM: '',
+      addressPVM: '',
+      addressCoreEth: '',
+    };
+
+    const provXP = isMainnet
+      ? Avalanche.JsonRpcProvider.getDefaultMainnetProvider()
+      : Avalanche.JsonRpcProvider.getDefaultFujiProvider();
+
+    try {
+      const publicKey = getPublicKeyFromPrivateKey(privateKey);
+      addresses.addressC = getEvmAddressFromPubKey(publicKey);
+      addresses.addressBTC = getBtcAddressFromPubKey(
+        publicKey,
+        isMainnet ? networks.bitcoin : networks.testnet
+      );
+      addresses.addressAVM = provXP.getAddress(publicKey, 'X');
+      addresses.addressPVM = provXP.getAddress(publicKey, 'P');
+      addresses.addressCoreEth = provXP.getAddress(publicKey, 'C');
+    } catch (err) {
+      throw new Error('Error while calculating addresses');
+    }
+
+    if (
+      !addresses.addressC ||
+      !addresses.addressBTC ||
+      !addresses.addressAVM ||
+      !addresses.addressPVM ||
+      !addresses.addressCoreEth
+    ) {
+      throw new Error(`Missing address`);
+    }
+
+    return addresses;
+  }
+
+  async addAddress({
+    index,
+    walletId,
+    ledgerService,
+    networkService,
```

Comment:
Can we inject `LedgerService` and `NetworkService` in the constructor instead of passing them through the method parameters here?